SYCL/Plugin/level_zero_ext_intel_cslice.cpp

```diff
@@ -0,0 +1,136 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: env ZEX_NUMBER_OF_CCS=0:4 env ZE_DEBUG=1 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER --check-prefixes=CHECK-PVC
+
+// Requires: level_zero
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+template <typename RangeTy, typename ElemTy>
+bool contains(RangeTy &&Range, const ElemTy &Elem) {
+  return std::find(Range.begin(), Range.end(), Elem) != Range.end();
+}
+
+bool isPartitionableBy(device &Dev, info::partition_property Prop) {
+  return contains(Dev.get_info<info::device::partition_properties>(), Prop);
+}
+
+bool isPartitionableByCSlice(device &Dev) {
+  return isPartitionableBy(
+      Dev, info::partition_property::ext_intel_partition_by_cslice);
+}
+
+bool isPartitionableByAffinityDomain(device &Dev) {
+  return isPartitionableBy(
+      Dev, info::partition_property::partition_by_affinity_domain);
+}
+
+void test_pvc(device &d) {
+  std::cout << "Test PVC Begin" << std::endl;
+  // CHECK-PVC: Test PVC Begin
+  bool IsPVC = [&]() {
+    if (!d.has(aspect::ext_intel_device_id))
+      return false;
+    return (d.get_info<ext::intel::info::device::device_id>() & 0xff0) == 0xbd0;
+  }();
+  std::cout << "IsPVC: " << std::boolalpha << IsPVC << std::endl;
+  if (IsPVC) {
+
+    assert(isPartitionableByAffinityDomain(d));
+    assert(!isPartitionableByCSlice(d));
+    {
+      try {
+        std::ignore = d.create_sub_devices<
+            info::partition_property::ext_intel_partition_by_cslice>();
+        assert(false && "Expected an exception to be thrown earlier!");
+      } catch (sycl::exception &e) {
+        assert(e.code() == errc::feature_not_supported);
+      }
+    }
+
+    auto sub_devices = d.create_sub_devices<
+        info::partition_property::partition_by_affinity_domain>(
+        info::partition_affinity_domain::next_partitionable);
+    device &sub_device = sub_devices[1];
+    assert(!isPartitionableByAffinityDomain(sub_device));
+    assert(isPartitionableByCSlice(sub_device));
+    assert(sub_device.get_info<info::device::partition_type_property>() ==
+           info::partition_property::partition_by_affinity_domain);
+
+    {
+      try {
+        std::ignore = sub_device.create_sub_devices<
+            info::partition_property::partition_by_affinity_domain>(
+            info::partition_affinity_domain::next_partitionable);
+        assert(false && "Expected an exception to be thrown earlier!");
+      } catch (sycl::exception &e) {
+        assert(e.code() == errc::feature_not_supported);
```

Comment:
Same here ... I'm not sure we want this to fail because existing code may depend on this working.